apps/crn-server/src/dependencies/external-authors.dependencies.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  getGraphQLClient as getContentfulGraphQLClient,
+  getRestClient as getContentfulRestClient,
+} from '@asap-hub/contentful';
+import {
+  RestExternalAuthor,
+  SquidexGraphql,
+  SquidexRest,
+} from '@asap-hub/squidex';
+import {
+  appName,
+  baseUrl,
+  contentfulAccessToken,
+  contentfulEnvId,
+  contentfulManagementAccessToken,
+  contentfulSpaceId,
+  isContentfulEnabledV2,
+} from '../config';
+import { ExternalAuthorContentfulDataProvider } from '../data-providers/contentful/external-authors.data-provider';
+import {
+  ExternalAuthorDataProvider,
+  ExternalAuthorSquidexDataProvider,
+} from '../data-providers/external-authors.data-provider';
+import { getAuthToken } from '../utils/auth';
+
+export const getExternalAuthorDataProvider = (): ExternalAuthorDataProvider => {
+  if (isContentfulEnabledV2) {
+    const contentfulGraphQLClient = getContentfulGraphQLClient({
+      space: contentfulSpaceId,
+      accessToken: contentfulAccessToken,
+      environment: contentfulEnvId,
+    });
+
+    const getContentfulRestClientFactory = () =>
+      getContentfulRestClient({
```

Comment:
It feels like there's a case here for allowing data providers to be read-only and so not provide a rest client at all. The reason there's no coverage here is because this argument is never used in this context, so it's possibly better to not provide it at all.